handler.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+const AWS = require('aws-sdk');
+const ccxt = require('ccxt');
+const axios = require('axios');
+
+function publishSNS(result) {
+  const topicArn = process.env.NOTIFY_TOPIC_ARN || 'arn:aws:sns:ap-southeast-1:156969275739:NOTIFY_CLIENT_APP';
+
+  const sns = new AWS.SNS();
+  const snsParams = {
+    Message: JSON.stringify(result),
+    Subject: "SNS from Transaction Lambda",
+    TopicArn: topicArn
+  };
+  sns.publish(snsParams, context.done);
+}
+
+module.exports.transaction = async (event, context) => {
+  const actionSNS = event.Records[0].Sns.TopicArn;
+
+  // SNS has single record only https://stackoverflow.com/questions/33690231/when-lambda-is-invoked-by-sns-will-there-always-be-just-1-record
+  // requires jobId in Message
+  const jobId = event.Records[0].Sns.Message;
+  const serviceBase = process.env.KEY_SERVICE_BASE_URL || 'http://pair.service.thinblock.io/';
+  const jobBase = process.env.JOBS_SERVICE_BASE_URL || 'http://jobs.service.thinblock.io/';
+  const result = {
+    jobId,
+    event: '',
+    data: ''
+  }
+
+  const jobURL = `${jobBase}jobs/${jobId}?look_up=_id`;
+  const { data: { actions } } = await axios.get(jobURL);
+
+  // find appropriate action in job
+  let actionIndex = -1;
+  actions.forEach((element, index) => {
+    if (element.action.sns_topic_arn === actionSNS) {
+      actionIndex = index;
+    }
+  });
+
+  if (actionIndex = -1) {
+    // Couldn't find appropriate action in job
+    result.event = null;
+    result.data = 'transaction_failed: Action not found in job';
```

Comment:
@abdullahtariq91 I think this is a good time to standardise the notification response. I think we should have `status` in the data and it should be `success|failed` so that the users can do something on their own with this data.
Also, we should give them an `error_message` in case of `failed` status. What do you think?
So to summarise:
```javascript
result.data = {
  status: 'failed',
  error_message: 'transaction_failed: action not found in job'
}
```